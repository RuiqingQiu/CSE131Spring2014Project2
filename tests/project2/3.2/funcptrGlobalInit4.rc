function : void foo() { 
   static int i;
   if (i == 0){ cout << "fe" << endl; }
   if (i == 1){ cout << "fi" << endl; }
   if (i == 2){ cout << "fo" << endl; }
   if (i == 3){ cout << "fum" << endl; i = -1; }
   i++;
}

auto fp = foo;
funcptr : void () fp4 = foo;
funcptr : void () fp7;

static auto fp10 = foo;
static funcptr : void () fp11 = foo;
static funcptr : void () fp12;

function : int main(){
   static auto fp2 = foo;
   static funcptr : void () fp5 = foo;
   
   auto fp3 = foo;
   funcptr : void () fp6 = foo;

   foo();
   fp();
   fp2();
   fp3();
   cout << endl;

   fp4();
   fp5();
   fp6();
   foo();
   cout << endl;

   static funcptr : void () fp8;
   funcptr : void () fp9;
   
   if (!(fp7 == nullptr)){ cout << "failure7" << endl;}
   if (fp8 != nullptr){ cout << "failure8" << endl;}

   fp7 = fp8 = fp9 = fp6;
   
   fp6();
   fp7();
   fp8();
   fp9();
   cout << endl;
   
   if (fp12 != nullptr){ cout << "failure12" << endl;}
   fp12 = fp11;
   fp12();
   fp10();
   fp11();
   foo();

   return 0;
}
