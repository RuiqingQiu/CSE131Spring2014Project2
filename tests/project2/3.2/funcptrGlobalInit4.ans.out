fe
fi
fo
fum

fe
fi
fo
fum

fe
fi
fo
fum

fe
fi
fo
fum
